### PR TITLE
feat(widget-configurator): use @cowprotocol/widget-react for configurator

### PR DESCRIPTION
**Pull request by shoom3301**
# Summary

Since the widget configurator is a react app, it 100% makes sense to use `@cowprotocol/widget-react` here

### CODE REVIEW

File: libs/widget-react/README.md [hunk 1]
```diff
@@ -30,12 +29,9 @@ const cowSwapWidgetParams: CowSwapWidgetParams = {
...
 const cowSwapWidgetSettings: CowSwapWidgetSettings = {
-  urlParams: {
-    chainId: 1,
-    tradeType: 'swap',
-    env: 'local',
-  },
-  appParams: {},
+  chainId: 1,
+  tradeType: 'swap',
+  env: 'local',
```
Comment (anxolin) [inline libs/widget-react/README.md:34]: maybe i wouldn't add the env in the basic example. Maybe not even the trade type. The simpler the better.

Comment (anxolin) [reply libs/widget-react/README.md:34]: do we still make the distinction params/setting ?

File: apps/widget-configurator/src/app/configurator/index.tsx [hunk 2]
```diff
@@ -72,28 +69,9 @@ export function Configurator({ title }: { title: string }) {
...
-      updateWidgetRef.current(settings)
-    } else {
-      updateWidgetRef.current = cowSwapWidget(params, settings)
-    }
-  }, [provider, params, settings])
-
-  useEffect(() => {
-    providerRef.current = provider
-  }, [provider])
-
```
Comment (anxolin) [inline apps/widget-configurator/src/app/configurator/index.tsx:96]: nice! 👋

**Review by anxolin — APPROVED** [review]
Great!

File: libs/widget-react/README.md (hunk 1 above)
Comment (shoom3301) [reply libs/widget-react/README.md:34]: Yes, I only merged `appParams` and `urlParams` into `CowSwapWidgetSettings`. But I would keep distinction between `CowSwapWidgetParams` and `CowSwapWidgetSettings`.
`CowSwapWidgetParams` contains static data like a `container`, `metaData` and `provider`, whereas `CowSwapWidgetSettings` is full of parameters that might me changed on flight.

Comment (shoom3301) [reply libs/widget-react/README.md]: >maybe i wouldn't add the env in the basic example

Makes sense. Made `CowSwapWidgetSettings` fully optional

**Pull request merged**